### PR TITLE
Add sea-ice graph partitioning to `files_for_e3sm` test case

### LINKED ISSUE #474: Use sea-ice load balanced partitioning in `files_for_e3sm`
Rather than naive partitioning, use the specialized tool from MPAS-Tools to do sea-ice partitioning:
https://github.com/MPAS-Dev/MPAS-Tools/tree/master/mesh_tools/seaice_grid_tools

This will require that these tools be available in the `mpas_tools` conda package, see:
https://github.com/MPAS-Dev/MPAS-Tools/issues/480

### PR DESCRIPTION
**Pull request by xylar**
This merge also generalizes `files_for_e3sm` so it can be run with an existing mesh and graph file, not just one generated by compass.  Thus, it replaces `make_diagnostics_files`, which performed a subset of this functionality for an existing mesh.

Checklist
* [x] User's Guide has been updated
* [x] Developer's Guide has been updated
* [x] API documentation in the Developer's Guide (`api.rst`) has any new or modified class, method and/or functions listed
* [x] Documentation has been [built locally](https://mpas-dev.github.io/compass/latest/developers_guide/building_docs.html) and changes look as expected
* [x] Document (in a comment titled `Testing` in this PR) any testing that was used to verify the changes

Closes #474

### CODE REVIEW

**Comment by xylar** [comment]
This will require a new version of `mpas_tools` with https://github.com/MPAS-Dev/MPAS-Tools/pull/482, https://github.com/MPAS-Dev/MPAS-Tools/pull/484 and https://github.com/MPAS-Dev/MPAS-Tools/pull/485

**Comment by xylar** [comment]
## Plotting

I plotted the QU240 (not shown) and SORRM partitions for several processor counts.  Below are images for 96 processors in the northern and southern hemispheres:
north
south

Given the higher resolution around Antarctica, it makes sense that most processors are allocated there.  It is also clear that each processor are being allocated a polar and an equatorial region, as expected.

**Comment by xylar** [comment]
@akturner, could you review these results either just using the above images or perhaps by taking a look at the following output directories?
```
/lcrc/group/e3sm/ac.xylar/compass_1.2/chrysalis/test_20230117/seaice_part_qu240_for_e3sm/ocean/global_ocean/QU240/PHC/files_for_e3sm/seaice_graph_partition
/lcrc/group/e3sm/ac.xylar/compass_1.2/chrysalis/test_20230117/seaice_part_sorrm_files_for_e3sm/
ocean/global_ocean/files_for_e3sm/seaice_graph_partition
```

File: compass/ocean/tests/global_ocean/files_for_e3sm/seaice_graph_partition.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+                '-o', '.']
+        check_call(args, logger)
+
+        args = ['create_seaice_partitions',
+                '-m', 'restart.nc',
+                '-o', '.',
+                '-p', f'mpas-seaice.graph.info.{creation_date}',
+                '-g', 'gpmetis',
+                '--plotting', 
+                '-n']
```
Comment (xylar) [inline compass/ocean/tests/global_ocean/files_for_e3sm/seaice_graph_partition.py:81]: @akturner, just a note that `--plotting` is always used here so we'll have the plot files to look at whenever we want them.

**Review by akturner — APPROVED** [review]
Approved based on examination of output partition diagnostic files

**Comment by xylar** [comment]
Thanks @akturner!  Very much appreciated.

**Pull request merged**